CRM/Export/BAO/Export.php

```diff
@@ -464,7 +463,7 @@ public static function exportComponents(
         $componentDetails[] = $row;
 
         // output every $tempRowCount rows
-        if ($count % $tempRowCount == 0) {
+        if (($count > 0) && ($count % $tempRowCount == 0)) {
```

Comment:
`$count % $tempRowCount == 0` evaluates to true when $count=0 but we don't want it to as that causes additional writes to the database and bypasses the batch writing.